src/Symfony/Component/Translation/Translator.php

```diff
@@ -214,7 +223,19 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
             }
         }
 
-        return strtr($this->selector->choose($catalogue->get($id, $domain), (int) $number, $locale), $parameters);
+        return $this->getFormatter($formatter)->format($locale, $this->getCatalogue($locale)->get((string) $id, $domain), $parameters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transChoice($id, $number, array $parameters = array(), $domain = null, $locale = null)
+    {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 3.1 and will be removed in 4.0. Use trans instead.', E_USER_DEPRECATED);
+
+        $parameters['__number__'] = $number;
```

Comment:
One possible BC break: we need to preserve `number` argument for the transChoice